ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -0,0 +1,166 @@
+import { Provider } from "../providers";
+import * as Gen from "../generated/index";
+
+const ansContractsMap: Record<string, string> = {
+  testnet: "0x5f8fd2347449685cf41d4db97926ec3a096eaf381332be4f1318ad4d16a8497c",
+  mainnet: "0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c",
+};
+
+// Each name component can only have lowercase letters, number or hyphens, and cannot start or end with a hyphen.
```

Comment:
it was taken from the ANS repo (not move code), the plan is that once we have it in the sdk, it would be removed from the ANS repo.
Other than that, not completely sure what you ask - this comment explains the regex we match to

Reply:
If I'm not wrong ANS must have effectively the same regex in the Move contract, right? That's code duplication, but we can't reasonably get rid of it, so let's at least have a comment pointing to the code. That way we'll know the change both if we change something.